docs/docs/reference/singleton-types.md

```diff
@@ -0,0 +1,39 @@
+---
+layout: doc-page
+title: "Literal Singleton Types"
+---
+
+Singleton types is one of the most popular features in Scala. Literal Singleton types takes it a step further.
+Right now, a popular way to replicate literal singleton types is to create macros and/or use libraries such as
+`shapeless` due to the lack of a syntax to express them.
+
+With Literal Singelton types, it is possible to do something like
+
+```scala
+val t: "Jedi".type = "Jedi"
+```
+or
+```scala
+val t: 42.type = 42
+```
+or
+```scala
+val t: 42 = 42
+```
+
+to use literal Singleton types. Trying to do the same in Scala produces the following error:
+```
+<console>:1: error: identifier expected but string literal found.
+val t: "Jedi".type = "Jedi"
+```
+
+Dotty provides support for using Literal Singleton types of the last format namely:
+```scala
+val t: 42 =42
+val x: "Jedi" = "Jedi"
+```
+and so on.
+
+There is a possibility to use singleton types in some contexts, but only on identifiers which points to a constant that conforms to `AnyRef`. This restriction is due to `Any` not having an eq method, which is what’s used for singleton type-equality check and pattern matching.
```

Comment:
I'm assuming you are talking about type contexts here? I'm not aware of this restriction with `AnyRef`, do you have an example? This seems to work fine:

```scala
scala> def f[T <: Any](t: T) = t
def f[T <: Any](t: T): [T](t: T)T

scala> f[1.1](1.1)
val res3: Double = 1.1
```